src/lib/services/universal-profile.service.ts

```diff
@@ -507,16 +584,25 @@ export function prepareTransferOwnershipTransaction$(
 
 export function isSignerUniversalProfile$(signer: Signer) {
   return defer(async () => {
-    try {
-      const signerAddress = await signer.getAddress();
-      const universalProfile = UniversalProfile__factory.connect(signerAddress, signer);
+    const signerAddress = await signer.getAddress();
+    return await addressIsUniversalProfile(signerAddress, signer);
+  }).pipe(shareReplay());
+}
+
+export async function addressIsUniversalProfile(address: string, signer: Signer) {
+  try {
+    const universalProfile = UniversalProfile__factory.connect(address, signer);
 
-      const owner = await universalProfile.owner();
-      return !!owner;
-    } catch (error) {
-      return false;
+    let isUniversalProfile = await universalProfile.supportsInterface(ERC725_ACCOUNT_INTERFACE);
+
+    if (!isUniversalProfile) {
+      isUniversalProfile = await universalProfile.supportsInterface('0x63cb749b');
     }
-  }).pipe(shareReplay());
+
+    return isUniversalProfile;
+  } catch (error) {
+    return false;
```

Comment:
maybe you can console.log or console.error the error so we get the opportunity to know that something went wrong?